examples/jetstream/limits-stream/node/main.js

```diff
@@ -0,0 +1,97 @@
+import {connect, StringCodec} from "nats";
+
+// Get the passed NATS_URL or fallback to the default. This can be
+// a comma-separated string.
+const servers = process.env.NATS_URL || "nats://localhost:4222";
+
+// Create a client connection to an available NATS server.
+const nc = await connect({
+  servers: servers.split(","),
+});
+
+// NATS message payloads are byte arrays, so we need to have a codec
+// to serialize and deserialize payloads in order to work with them.
+// Another built-in codec is JSONCodec or you can implement your own.
+const sc = StringCodec();
+
+// Access the JetStream manager which provides the methods for managing
+// streams and consumers.
+const jsm = await nc.jetstreamManager();
+
+// Declare the initial stream config. A stream can bind one or more
+// subjects that are not overlapping with other streams. By default,
+// a stream will have one replica and use file storage.
+const cfg = {
+  name: "EVENTS",
+  subjects: ["events.>"],
+};
+
+// Add/create the stream.
+await jsm.streams.add(cfg)
+console.log("created the stream")
+
+// Access the JetStream client for publishing and subscribing to streams.
+const js = nc.jetstream();
+
+// Publish a series of messages and wait for each one to be completed.
+await js.publish("events.page_loaded");
+await js.publish("events.mouse_clicked");
+await js.publish("events.mouse_clicked");
+await js.publish("events.page_loaded");
+await js.publish("events.mouse_clicked");
+await js.publish("events.input_focused");
+console.log("published 6 messages");
+
+const events = [
+  "events.input_changed",
+  "events.input_blurred",
+  "events.key_pressed",
+  "events.input_focused",
+  "events.input_changed",
+  "events.input_blurred",
+];
+
+// Publish a series of events and wait for _all_ of the publishes
+// have resolved.
+const batch = [];
+events.forEach((e) => {
+  batch.push(js.publish(e));
+})
+
+await Promise.all(batch);
+console.log("published another 6 messages");
+
+// Get the stream state to show 12 messages exist.
+let info = await jsm.streams.info(cfg.name);
+console.log(info.state);
+
+// Let's update the stream config and set the max messages to 10. This
+// can be done with a partial config passed to the `update` method.
+await jsm.streams.update(cfg.name, {max_msgs: 10});
+
+// Once applied, we can check out the stream state again to see that the
+// first two messages were truncated.
+info = await jsm.streams.info(cfg.name);
+console.log(info.state);
+
+// Updating the stream config again, we can set the max bytes of the stream
+// overall.
+await jsm.streams.update(cfg.name, {max_bytes: 300});
+
+// This will prune the messages down some more...
+info = await jsm.streams.info(cfg.name);
+console.log(info.state);
+
+// Finally the last limit of max_age can be applied. Note the age
+// is in nanoseconds, so 1 billion => 1 second.
+await jsm.streams.update(cfg.name, {max_age: 1000000000});
+
+// Sleep for a second.
+await new Promise(r => setTimeout(r, 1000));
```

Comment:
not quite sure why you are adding the delay - the update should be completed by the time the update returns - it should have the correct config. But perhaps you do not see the same thing.

Reply:
Without sleeping, it it appears the next info state still shows the messages in the stream. Everything may be happening so quick that a max age of a second is still too long.

Reply:
for easier code reading I would also change the line to be, this will explicitly point out that we are dealing with nanos.

await jsm.streams.update(cfg.name, {max_age: nanos(1000)});